### DISCUSSION TITLE
Changing settings

### DISCUSSION THREAD

**Discussion by leonatkins** [ORelio/Minecraft-Console-Client#2177] [Q&A]
When I change settings, how do I make them active? I'm on Mac

**Comment by milutinke** [comment]
> When I change settings, how do I make them active? I'm on Mac

Currently you need to restart the client.
I made a `/reload` command, I need testers, read this: https://github.com/MCCTeam/Minecraft-Console-Client/pull/2167#issuecomment-1242759750

**Reply by leonatkins** [reply]
How do I restart it?

**Reply by milutinke** [reply]
> How do I restart it?

Just close it and run it again.

**Reply by leonatkins** [reply]
> > How do I restart it?
> 
> Just close it and run it again.

But I did and I still don't have InventoryHandling enabled.

**Reply by milutinke** [reply]
> > > How do I restart it?
> > 
> > 
> > Just close it and run it again.
> 
> But I did and I still don't have InventoryHandling enabled.

Which version are you using?
The inventory handling is not implemented for versions bellow 1.15.

Also, check if you are modifying a right configuration file, best option is to delete all of them and let MCC generate a new one and then edit it.

**Reply by leonatkins** [reply]
> > > > How do I restart it?
> > > 
> > > 
> > > Just close it and run it again.
> > 
> > 
> > But I did and I still don't have InventoryHandling enabled.
> 
> Which version are you using? The inventory handling is not implemented for versions bellow 1.15.
> 
> Also, check if you are modifying a right configuration file, best option is to delete all of them and let MCC generate a new one and then edit it.

I don't know which version. The path to the config file is Minecraft-Console-Client/MinecraftClient/Resources/config.

**Reply by leonatkins** [reply]
I've also set my login email in the settings but it still asks me for it.

**Reply by milutinke** [reply]
You're editing the original file from the source, not the one that the MCC generates, unless you're specifying it to be loaded via the command line parameter.
The MCC auto generates MinecraftClient.ini in the same folder where it's located, you should edit it.

PS: Accidentally edited your reply instead of replying xD, fixed.

**Reply by leonatkins** [reply]
> You're editing the original file from the source, not the one that the MCC generates, unless you're specifying it to be loaded via the command line parameter. The MCC auto generates MinecraftClient.ini in the same folder where it's located, you should edit it.
> 
> PS: Accidentally edited your reply instead of replying xD, fixed.

Can I get the file path? I don't know what you are talking about.

**Reply by milutinke** [reply]
> > You're editing the original file from the source, not the one that the MCC generates, unless you're specifying it to be loaded via the command line parameter. The MCC auto generates MinecraftClient.ini in the same folder where it's located, you should edit it.
> > PS: Accidentally edited your reply instead of replying xD, fixed.
> 
> Can I get the file path? I don't know what you are talking about.

When you run the `MinecraftConsoleClient` program, it will auto create it's configuration file in the same folder the program is located, it's name will be `MinecraftClient.ini`, you should edit that file.

**Reply by milutinke** [reply]
![image](https://user-images.githubusercontent.com/441903/189497536-c6584d90-73ed-437c-ae3e-9fa3c240be50.png)

**Reply by leonatkins** [reply]
> > > You're editing the original file from the source, not the one that the MCC generates, unless you're specifying it to be loaded via the command line parameter. The MCC auto generates MinecraftClient.ini in the same folder where it's located, you should edit it.
> > > PS: Accidentally edited your reply instead of replying xD, fixed.
> > 
> > 
> > Can I get the file path? I don't know what you are talking about.
> 
> When you run the `MinecraftConsoleClient` program, it will auto create it's configuration file in the same folder the program is located, it's name will be `MinecraftClient.ini`, you should edit that file.

The only MinecraftClient.ini file is in `Resources/config`

**Reply by leonatkins** [reply]
![image](https://user-images.githubusercontent.com/102980775/189497574-9d5441d9-430a-4ac1-929f-b1132dd2e930.png)

**Reply by milutinke** [reply]
Oh, you've built it from the source.
Ok, run it, and it will make the ini file, then exit it and edit the newly created file.

**Reply by leonatkins** [reply]
> Oh, you've built it from the source.
> Ok, run it, and it will make the ini file, then exit it and edit the newly created file.

I ran it but it didn't create the file.

**Reply by milutinke** [reply]
> > Oh, you've built it from the source.
> > Ok, run it, and it will make the ini file, then exit it and edit the newly created file.
> 
> I ran it but it didn't create the file.

Did it start correctly?
Also, can you refresh the folder and open a terminal it it and type: `ls -laH` and screenshot it.

**Reply by milutinke** [reply]
https://user-images.githubusercontent.com/441903/189497753-230a996d-8aa1-43e7-a733-5fc9945e38c4.mp4

**Reply by leonatkins** [reply]
That didn't happen, it's still the same.

**Reply by milutinke** [reply]
Does the client even run?
Do you get the login prompt?

**Reply by leonatkins** [reply]
> Does the client even run?
> Do you get the login prompt?

Yes

**Reply by milutinke** [reply]
Weird.
Maybe it does not have permissions to create a new file.
Try doing `chmod 777 MinecraftClient` and then try running it.

**Reply by leonatkins** [reply]
> Weird.
> Maybe it does not have permissions to create a new file.
> Try doing `chmod 777 MinecraftClient` and then try running it.

Ok, just did it. It didn't say anything when I did it and it still isn't working.

**Reply by milutinke** [reply]
No idea why that is happening, is there an error maybe?
You could use the one file you've edited by starting the program from the terminal and providing a full path to the file you've edited.

Usage:
```bash
./MinecraftClient <full path to the ini file>
```

Example:
```bash
./MinecraftClient /home/someuser/Minecraft-Console-Client/MinecraftClient/Resources/config/MinecraftClient.ini
```

**Reply by leonatkins** [reply]
> No idea why that is happening, is there an error maybe? You could use the one file you've edited by starting the program from the terminal and providing a full path to the file you've edited.
> 
> Usage:
> 
> ```shell
> ./MinecraftClient <full path to the ini file>
> ```
> 
> Example:
> 
> ```shell
> ./MinecraftClient /home/someuser/Minecraft-Console-Client/MinecraftClient/Resources/config/MinecraftClient.ini
> ```

`zsh: permission denied: ./MinecraftClient`

**Reply by milutinke** [reply]
Do: `chmod +x MinecraftClient` and then run it.

**Reply by leonatkins** [reply]
> Do: `chmod +x MinecraftClient` and then run it.

You are asking me to run a folder, which I can't do.

**Reply by milutinke** [reply]
Not a folder, but the program, it's called MinecraftClient, doesn't have an extension on Linux/MacOS.
Navigate to the folder where the program is and run it.

**Reply by leonatkins** [reply]
No I'm pretty sure it's a folder.
![image](https://user-images.githubusercontent.com/102980775/189499960-ee62ac9f-10e7-43ce-b663-43641fa7a034.png)

**Reply by leonatkins** [reply]
> Not a folder, but the program, it's called MinecraftClient, doesn't have an extension on Linux/MacOS.
> Navigate to the folder where the program is and run it.

Also still permission denied.

**Reply by milutinke** [reply]
Which System are you running, looks like a linux distro?
Since you've built from source, you should go to: `MinecraftClient/bin/Release/linux-64/publish`
(If it is a macOS, the folder inside Release will be /osx-64)
(It you're on an ARM cpu, then it's going to be linux-arm64)

Or simply you can download a release from here: https://github.com/MCCTeam/Minecraft-Console-Client/releases/tag/20220909-54
And put it wherever you like and run it, it should have executable permission by default.

**Reply by leonatkins** [reply]
> Which System are you running, looks like a linux distro? Since you've built from source, you should go to: `MinecraftClient/bin/Release/linux-64/publish` (If it is a macOS, the folder inside Release will be /osx-64) (It you're on an ARM cpu, then it's going to be linux-arm64)
> 
> Or simply you can download a release from here: https://github.com/MCCTeam/Minecraft-Console-Client/releases/tag/20220909-54

I'm on Mac.

**Reply by milutinke** [reply]
You can download pre-built one, put it wherever you prefer and then run it.
The link is above.

**Reply by leonatkins** [reply]
> Which System are you running, looks like a linux distro? Since you've built from source, you should go to: `MinecraftClient/bin/Release/linux-64/publish` (If it is a macOS, the folder inside Release will be /osx-64) (It you're on an ARM cpu, then it's going to be linux-arm64)
> 
> Or simply you can download a release from here: https://github.com/MCCTeam/Minecraft-Console-Client/releases/tag/20220909-54 And put it wherever you like and run it, it should have executable permission by default.

It didn't create any files.

**Reply by milutinke** [reply]
No idea why that is happening, try giving it permissions now and then trying.
If it does not work, then copy over the default MinecraftClient.ini from this link: https://raw.githubusercontent.com/MCCTeam/Minecraft-Console-Client/master/MinecraftClient/Resources/config/MinecraftClient.ini

**Reply by leonatkins** [reply]
How do I give it permissions again?

**Reply by milutinke** [reply]
`chmod +x MinecraftClient`

**Reply by leonatkins** [reply]
> chmod +x MinecraftClient

Is it supposed to respond with anything?

**Reply by milutinke** [reply]
Nope, just gives a blank output.

**Reply by leonatkins** [reply]
> Nope, just gives a blank output.

I gave it permissions, still doesn't work

**Reply by milutinke** [reply]
> > Nope, just gives a blank output.
> 
> I gave it permissions, still doesn't work

Then copy over the default .ini file to the location where the program is.

**Reply by leonatkins** [reply]
> > > Nope, just gives a blank output.
> > 
> > 
> > I gave it permissions, still doesn't work
> 
> Then copy over the default .ini file to the location where the program is.

I did and I enabled InventoryHandling and added my login email but it still prompted me for it and says InventoryHandling isn't on.

**Reply by milutinke** [reply]
Which version are you playing on?

**Reply by leonatkins** [reply]
> Which version are you playing on?

How do I check?

**Reply by milutinke** [reply]
> > Which version are you playing on?
> 
> How do I check?

When you connect to a server it should tell you
Example: `Server version : Paper 1.19.2 (protocol v760)`

**Reply by leonatkins** [reply]
It says `Server version : Requires MC 1.8 - 1.19 (protocol v760)`

**Reply by milutinke** [reply]
> It says `Server version : Requires MC 1.8 - 1.19 (protocol v760)`

If the server is bellow 1.15 then the inventory handling is not implemented.
It is a public server, if so, can you give me an ip so I could do some testing?

**Reply by leonatkins** [reply]
`play.lemoncloud.com`

**Reply by milutinke** [reply]
> play.lemoncloud.com

That ip doesn't exist, can't connect.

**Reply by Splarkszter** [reply]
if you go to lemoncloud.com on the browser it shows a hosting website. Could be a private server that he manages and could be currently off.

**Reply by Splarkszter** [reply]
Or maybe his server is not even set up yet.

**Reply by leonatkins** [reply]
Whoops, I meant `play.lemoncloud.org` and no, it's not my server.

**Reply by milutinke** [reply]
> Whoops, I meant `play.lemoncloud.org` and no, it's not my server.

Tested the terrain movement there, on the hub and it works.

**Reply by leonatkins** [reply]
> > Whoops, I meant `play.lemoncloud.org` and no, it's not my server.
> 
> Tested the terrain movement there, on the hub and it works.

The settings don't work though.

**Reply by leonatkins** [reply]
Nevermind! I got it all working!
